loadtest/control/browsercontroller/controller.go

```diff
@@ -0,0 +1,294 @@
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+)
+
+// This is the URL of the LTBrowser API server ran from /browser
+const LTBrowserApi = "http://localhost:5000"
+
+// BrowserController is a controller that manages browser sessions
+// by communicating with the browserLt API server.
+type BrowserController struct {
+	id              int
+	user            user.User
+	status          chan<- control.UserStatus
+	stopChan        chan struct{}
+	stoppedChan     chan struct{}
+	wg              *sync.WaitGroup
+	ltBrowserApiUrl string
+	httpClient      *http.Client
+}
+
+// AddBrowserRequest represents the request body for adding a browser session
+type AddBrowserRequest struct {
+	// User is the username or email of the user to add a browser session
+	User     string `json:"user"`
+	Password string `json:"password"`
+}
+
+// BrowserAPIResponse represents the response from the browser API
+type BrowserAPIResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+	Error   *struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error,omitempty"`
+}
+
+// New creates and initializes a new BrowserController with given parameters.
+func New(id int, user user.User, status chan<- control.UserStatus) (*BrowserController, error) {
+	if user == nil {
+		return nil, errors.New("user cannot be nil")
+	}
+	if status == nil {
+		return nil, errors.New("status channel cannot be nil")
+	}
+
+	return &BrowserController{
+		id:              id,
+		user:            user,
+		status:          status,
+		stopChan:        make(chan struct{}),
+		stoppedChan:     make(chan struct{}),
+		wg:              &sync.WaitGroup{},
+		ltBrowserApiUrl: LTBrowserApi,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}, nil
+}
+
+// Run starts the browser controller by creating a browser session.
+// This corresponds to adding a user in the browser load test.
+func (c *BrowserController) Run() {
+	if c.user == nil {
+		c.sendFailStatus("browser controller was not initialized")
+		return
+	}
+
+	c.status <- control.UserStatus{
+		ControllerId: c.id,
+		User:         c.user,
+		Info:         "browser controller started",
+		Code:         control.USER_STATUS_STARTED,
+	}
+
+	defer func() {
+		c.user.ClearUserData()
+		c.sendStopStatus()
+		close(c.stoppedChan)
+	}()
+
+	initActions := []control.UserAction{
+		control.SignUp,
+		control.Login,
+		control.JoinTeam,
+	}
+
+	for _, action := range initActions {
+		if resp := action(c.user); resp.Err != nil {
+			c.status <- c.newErrorStatus(resp.Err)
+			return
+		} else if resp.Info != "" {
+			c.status <- c.newInfoStatus(resp.Info)
+		}
+
+		// If stop signal is received then stop the controller
+		select {
+		case <-c.stopChan:
+			return
+		default:
+		}
+	}
+
+	if err := c.addBrowser(); err != nil {
+		c.status <- c.newErrorStatus(control.NewUserError(err))
+		return
+	}
+	c.status <- c.newInfoStatus("browser added successfully")
```

Comment:
I know we discussed this in our call, but don't remember our decision: are we keeping the `control` actions here and remove them in a future PR, or are we removing them now? We should be aware that this essentially creates two sessions: one controlled by the `control` functions, another controlled by the browser.

Reply:
I removed the login and join team method and moved them to simulation tests so we no longer have the duplicate sessions anymore